src/main/java/com/avaya/letscode/Cart.java

```diff
@@ -33,13 +33,19 @@ public int getQuantity() {
 
     public double getCartValue() {
 
-        double cartValue = this.items.values().stream().mapToInt(CartItem::getCartItemValue).sum();
-        //stream offer list and apply each offer to the cart
-        double discount = this.cartOffers.stream().mapToDouble(offer -> offer.getDiscountPercentage(this)).sum();
-        return cartValue - cartValue*discount/100;
+        //stream offer list and apply each offer to the cart and find the minimum value
+        // if cart has no offers, return the cart value before offer
+        if (this.cartOffers.isEmpty()) 
+            return this.getCartValueBeforeOffer();
+        else
+            return this.cartOffers.stream().mapToDouble(offer -> offer.getDiscountedValue(this)).min().getAsDouble();
         
     }
 
+    public double getCartValueBeforeOffer() {
```

Comment:
Can we call this simply getValueBeforeOffer() as we are already inside the Cart object. This will be on consistent lines with getQuantity().